idaes/power_generation/properties/natural_gas_PR.py

```diff
@@ -0,0 +1,465 @@
+
+from idaes.generic_models.properties.core.pure import NIST
+from idaes.generic_models.properties.core.pure import RPP4
+
+from idaes.generic_models.properties.core.reactions.dh_rxn import \
+    constant_dh_rxn
+from idaes.generic_models.properties.core.reactions.rate_constant import \
+    arrhenius
+from idaes.generic_models.properties.core.reactions.rate_forms import \
+    power_law_rate
+from idaes.generic_models.properties.core.generic.generic_reaction import (
+    ConcentrationForm)
+
+# Set up logger
+_log = logging.getLogger(__name__)
+
+# Property Sources
+
+# Source: NIST webbook
+# Properties: Heat capacity coefficients for all species except ethane,
+# propane, and butane. Reference enthalpies and entropies for all species.
+
+# Source: The Properties of Gases and Liquids (1987)
+# 4th edition, Chemical Engineering Series - Robert C. Reid
+# Properties: Critical temperatures and pressures. Omega.
+# Heat capacity coefficients for ethane, propane, and butane.
+
+
+class enth_mol_ig_comp():
+
+    @staticmethod
+    def build_parameters(cobj):
+        if not hasattr(cobj, "cp_mol_ig_comp_coeff_A"):
+            NIST.cp_mol_ig_comp.build_parameters(cobj)
+
+    @staticmethod
+    def return_expression(b, cobj, T):
+        # Specific enthalpy via the Shomate equation
+        t = pyunits.convert(T, to_units=pyunits.kiloK)
+
+        h = (cobj.cp_mol_ig_comp_coeff_A*(t) +
+             (cobj.cp_mol_ig_comp_coeff_B/2)*(t**2) +
+             (cobj.cp_mol_ig_comp_coeff_C/3)*(t**3) +
+             (cobj.cp_mol_ig_comp_coeff_D/4)*(t**4) -
+             cobj.cp_mol_ig_comp_coeff_E*(1/t) +
+             cobj.cp_mol_ig_comp_coeff_F)
+
+        units = b.params.get_metadata().derived_units
+        return pyunits.convert(h, units["energy_mole"])
+
+
+# returns a configuration dictionary for the list of specified components
+def get_NG_properties(components='all'):
+    configuration = {
+        'components': {
+            'H2': {'type': Component,
+                   "elemental_composition": {"H": 2},
+                   'enth_mol_ig_comp': enth_mol_ig_comp,
+                   'entr_mol_ig_comp': NIST,
+                   'parameter_data': {
+                       'mw': (0.0020159, pyunits.kg/pyunits.mol),
+                       'pressure_crit': (13e5, pyunits.Pa),
+                       'temperature_crit': (33.2, pyunits.K),
+                       'omega': -0.218,
+                       'cp_mol_ig_comp_coeff': {'A': 33.066178,
+                                                'B': -11.363417,
+                                                'C': 11.432816,
+                                                'D': -2.772874,
+                                                'E': -0.158558,
+                                                'F': -9.980797,
+                                                'G': 172.707974,
+                                                'H': 0.0},
```

Comment:
Should we add the actual value for H from NIST here as well (even though it is not used)?

Reply:
The value of H for elemental species (H2, O2, N2, and Ar) is 0. The other species have the actual H values.